### PR TITLE
Add optional timestamps to cached HTML

### PR DESCRIPTION
**Pull request by DorsetDigital**
As per #78 
Add code to FilesystemPublisher to deal add timestamps

### CODE REVIEW

File: src/Publisher/FilesystemPublisher.php [hunk 1]
```diff
@@ -148,7 +167,12 @@ protected function publishPage($response, $url)
                 $phpContent = $this->generatePHPCacheFile($response);
                 $success = $this->saveToPath($phpContent, $path . '.php');
             }
-            return $this->saveToPath($response->getBody(), $path . '.html') && $success;
+            $responseBody = $response->getBody();
+            if ($this->getAddTimestamp() === true) {
+                $responseBody = str_replace('</html>', "<!-- " . date('c') . " -->\n</html>", $responseBody);
```
Comment (dhensby) [inline src/Publisher/FilesystemPublisher.php]: Can we use `DateTime::now()` please as this will be testable

**Comment by dhensby** [comment]
Thanks for this, how comes you're not using a config setting?

Please can you also add some docs for this (and squash your current commits)?

**Comment by DorsetDigital** [comment]
I figured it was more 'Stripey' to be able to set it in the yml file (similar to the file extension option).  Is that not the best way?  Happy to take guidance on this.

**Comment by dhensby** [comment]
> I figured it was more 'Stripey' to be able to set it in the yml file (similar to the file extension option). Is that not the best way? Happy to take guidance on this.

oh, setting it through injector params? I guess we could do it that way.

Another point though, can you make this change in `Publisher@generatePageResponse` and update the body of the response so that any/all publishers have this added rather than just the filesystempublisher?

**Comment by DorsetDigital** [comment]
I've moved the functionality into the Publisher class and swapped to a config-based approach, it seemed like a simpler way to do it with hindsight.
I'll add something to the docs now

**Comment by DorsetDigital** [comment]
I've now squashed all the commits as well, so hopefully I didn't break anything in the process!

File: src/Publisher.php [hunk 2]
```diff
@@ -92,6 +99,10 @@ public function generatePageResponse($url)
             );
             $app = $this->getHTTPApplication();
             $response = $app->handle($request);
+
+            if ($this->config()->get('add_timestamp') === true) {
+                $response->setBody(str_replace('</html>', "<!-- " . DBDateTime::now() . " -->\n</html>", $response->getBody()));
```
Comment (dhensby) [inline src/Publisher.php]: `DBDateTime::now()` needs to be formatted appropriately eg: `DBDateTime::now()->Full()` or maybe `DBDateTime::now()->Rfc2822()`

Comment (dhensby) [reply src/Publisher.php]: You also need to bring this line below 120 chars long

**Comment by DorsetDigital** [comment]
Are there still changes needed for this to be merged?   (I just wanted to make sure you're not waiting on me to do anything!)

**Comment by dhensby** [comment]
sorry, fell off my radar - looks acceptable to me

**Pull request merged**